crates/pop-drink/src/error.rs

```diff
@@ -0,0 +1,258 @@
+			},
+			RuntimeError::Raw(error) => decode::<T::PopApiError>(&error.encode()),
+		};
+		pop_api_error.into()
+	}
+}
+
+impl<T: RuntimeErrorConfig> From<u32> for RuntimeError<T> {
+	/// Converts a numerical value of `PopApiError` into a `RuntimeError`.
+	///
+	/// This is used to reconstruct and display a `RuntimeError` from its numerical representation
+	/// when an error is thrown.
+	fn from(value: u32) -> Self {
+		let error = T::PopApiError::from(value);
+		let encoded = error.encode();
+		if encoded[0] == T::MODULE_INDEX {
+			let (index, module_error) = (encoded[1], &encoded[2..]);
+			let data = vec![vec![index], module_error.to_vec()].concat();
+			return RuntimeError::Module(decode(&data));
+		}
+		RuntimeError::Raw(error)
+	}
+}
+
+pub mod v0 {
+	use std::fmt::Debug;
+	/// A utility macro to assert that an error returned from a smart contract method matches the
+	/// `RuntimeError`.
+	///
+	/// # Generic parameters:
+	///
+	/// - `R` - Success type returned if Ok().
+	/// - `E` - Returned `Err()` value of a method result. Must be convertable to `u32`.
+	/// - `RuntimeError` - Runtime error type.
+	///
+	/// # Parameters:
+	///
+	/// - `result` - Result returned by a smart contract method.
+	/// - `expected_error` - `RuntimeError` to be asserted.
+	pub fn assert_runtime_err_inner<R, E, RuntimeError>(
+		result: Result<R, E>,
+		expected_error: RuntimeError,
+	) where
+		E: Into<u32>,
+		RuntimeError: From<u32> + Into<u32> + Debug,
+	{
+		crate::error::assert_runtime_err_inner::<pop_primitives::Error, R, E, RuntimeError>(
+			result,
+			expected_error,
+		)
+	}
+
+	/// A utility macro to assert that an error returned from a smart contract method using Pop API
+	/// V0.
+	///
+	/// # Parameters:
+	///
+	/// - `result` - The result returned by a smart contract method. It is of type `Result<R, E>`,
+	///   where the error type `E` must implement a conversion to `u32`.
+	/// - `error` - A `RuntimeError` type configured specifically for the Pop API V0.
+	///
+	/// # Example:
+	///
+	/// ```rs
+	/// use drink::devnet::{
+	/// 	error::{
+	/// 		v0::{assert_runtime_error, RuntimeError},
+	/// 		Assets,
+	/// 		AssetsError::AssetNotLive,
+	/// 	}
+	/// };
+	///
+	/// /// Example `pop-drink` testing method to interact with PSP22 contract.
+	/// fn mint(session: &mut Session<Pop>, account: AccountId, amount: Balance) -> Result<(), PSP22Error> {
+	/// 	call::<Pop, (), PSP22Error>(
+	/// 		session,
+	/// 		"Psp22Mintable::mint",
+	/// 		vec![account.to_string(), amount.to_string()],
+	/// 		None,
+	/// 	)
+	/// }
+	///
+	/// /// Using macro to test the returned error.
+	/// assert_runtime_error!(
+	/// 	mint(&mut session, BOB, AMOUNT),
+	/// 	RuntimeError::Module(Assets(AssetNotLive))
+	/// );
+	/// ```
+	#[macro_export]
+	macro_rules! __assert_runtime_err {
+		($result:expr, $error:expr $(,)?) => {
+			$crate::error::v0::assert_runtime_err_inner::<_, _, _>($result, $error);
+		};
+	}
+
+	pub use __assert_runtime_err as assert_runtime_err;
```

Comment:
Assume this might be related to a naming clash, but would this not be better as `assert_err` due to the fact that runtime is implied by drink? The dev can choose which to import and alias as necessary, assuming that is indeed possible for macro imports.

Reply:
Resolved